### ISSUE TITLE
Menu in editor toolbar

### ISSUE THREAD

**Issue by Ragos81** [saschaleib/dokuwiki-plugin-adhocwrap#2]
I have replaced the wrap module with adhocwrap, but I'm missing the menu in the editor tool bar. Is there any chance that this could be added in a similar style like from the wrap plugin. I'm aware that adhocwrap can do more than, wrap, but it would be nice to have the "basic" functionality (like inserting info boxes) in the editor menu.

Unfortunately I'm not a programmer and have only very basic knowledge, otherwise I would do it myself and create a pull request.

**Comment by saschaleib** [comment]
The latest update of 10.03.2025 now has a menu with a selection of typical HTML phrases using the Ad-Hoc Wrap plugin styles.

I'm open to suggestions what should still be added there – but of course, it should be universally useful, and specifically use styles supplied by this plugin.

**Comment by Ragos81** [comment]
Great, thank you @saschaleib !
I think I can replace the wrap plugin with adhocwrap soon!

I tested all the buttons with the placeholder texts, and everything except the Illustration blocks works. 

![Image](https://github.com/user-attachments/assets/40579f21-54de-471e-8b6a-f3a6795c1bce)

Maybe I just missed a switch in the config?

![Image](https://github.com/user-attachments/assets/4fc7b8f2-51c8-4f45-8278-1edb2934b0f7)

I'm also wondering if the symbols for the Information-, Tip-, Important- and Alert box are intended to be so small, or if it's somehow influenced by an addon or the dokuwiki standard template.
If it's by design, is there any chance to increase the size without that it get's reset with the next adhocwrap update?

Besides that, should I close the issue and open another 2 for these points?

**Comment by Ragos81** [comment]
After checking some pages, I noticed that the problem with the illustration blocks came from the [latexcaption](https://www.dokuwiki.org/plugin:latexcaption) plugin. This give me 2 options, both of them are not great.

1. Replace latexcaption with adhoc, which would be preferred to minimize the number of plugins. But I would like to have the full functionality, as I also use it for tables and refer to a specific images or tables. I'm also not sure if adhoc can have subfigures.
2. Deactivate the ``<figure>`` and ``<figcaption>`` switch for adhoc. This would cause to have 2 "dead" buttons from the adhoc wrap plugin in the menu.

In this case, I guess I will have to choose option 2 and manually remove or hide the two buttons from the menu, unless you have a better idea.

**Comment by saschaleib** [comment]
Hello @Ragos81, from a first glance, it looks like the `<figure>` tag wasn't closed properly, and that's why it wasn't recognized. 

if there is another plugin interfering with the `<figure>` elements, then this is very likely the cause of the problems here. 

The only advise I can give in this case is to try if _latexcaption_ completely takes over the `<figure>` tags, if they are disabled in Ad-Hoc HTML.

When I find some time, I might have a closer look at that plugin and see how it works, but it seems very likely that this conflict can not be easily resolved, simply because of the way these plugins work... :-/

Edit: BTW: Since a recent update, buttons in the Plugin's popup only appear if the tags are actually active. So there would be no "dead buttons" in there, after all... :-)

**Comment by Ragos81** [comment]
I think I will stick to _latexcaption_ for the time being, as I don't have the need to switch. It does what it should, and I was just looking into the option to reduce the number of plugins.

> Edit: BTW: Since a recent update, buttons in the Plugin's popup only appear if the tags are actually active. So there would be no "dead buttons" in there, after all... :-)

I didn't know that, that's great!

**Comment by saschaleib** [comment]
Indeed, the reason why there is a selection of the tags you want to activate is just that you can avoid conflicts like this one – and use the plugin that best serves your needs for each element separately.

Let me just add one note: the Ad-Hoc HTML Plugin can see which elements are active in its settings, and only show the buttons in its menu for which the elements are actually active.

Other plugins – including the "Ad-Hoc Wrap"-Plugin, can not (easily) see that, and it is possible to disable elements (in AD-HTML) that some of the buttons referred to... thus indeed creating "dead" buttons. Just to keep that in mind :-)

Best greetings :-)
